presto-tests/src/test/java/com/facebook/presto/memory/TestMemoryManager.java

```diff
@@ -56,17 +99,72 @@ public void testClusterPools()
                 MILLISECONDS.sleep(100);
```

Comment:
you only want to sleep if the pool is null... you could do something like this
```
ClusterMemoryPool reservedPool;
while ((reservedPool = memoryManager.getPools().get(RESERVED_POOL)) == null) {
    MILLISECONDS.sleep(100);
}
```
Of course it has a yucky assignment in a condition, but it doesn't sleep if you got a non null.  Also, 100ms is a long time, I'd use 10ms.